ggml.c

```diff
@@ -21148,6 +21148,17 @@ static int gguf_get_or_add_key(struct gguf_context * ctx, const char * key) {
     return n_kv;
 }
 
+void gguf_remove_key(struct gguf_context * ctx, const char * key) {
+    const int idx = gguf_find_key(ctx, key);
+    if (idx >= 0) {
+        const int n_kv = gguf_get_n_kv(ctx);
+        for (int i = idx; i < n_kv; ++i)
+            ctx->kv[i] = ctx->kv[i+1];
+        ctx->kv = realloc(ctx->kv, (n_kv - 1) * sizeof(struct gguf_kv));
```

Comment:
The removed key and value may need to be freed as well.

Reply:
@slaren `realloc` will deallocate memory for removed key and value, since they are stored in the same struct

Reply:
Not for every type, check `gguf_free`.

Reply:
Added a common function to free key and value memory `gguf_free_kv`